subsys/net/ip/net_context.c

```diff
@@ -206,6 +216,14 @@ int net_context_get(sa_family_t family,
 			}
 		}
 
+#if defined(CONIFG_NET_TLS) || defined(CONFIG_NET_DTLS)
+		if (tls) {
+			if (net_tls_enable(&contexts[i], true) < 0) {
+				break;
```

Comment:
What error will be returned in this case? Is it good one?

Reply:
> -ENOENT, the same that in case there is no TCP context available.

It's separate question whether what's done for TCP context is good (we know that the current IP stack code is fully of questionable things if not outright bugs).

But enabling TLS is complex operation and might fail for different reasons, which reason should be communicated as a return value of net_tls_enable(). And I'd be very surprised if something else but that actual error should be returned to caller here.